providers/curation/github.js

```diff
@@ -139,44 +140,28 @@ class GitHubCurationService {
    * curated revision. The returned value will be decorated with a non-enumerable `_origin` property
    * indicating the sha of the commit for the curations if that info is available.
    */
-  getAll(coordinates, pr = null) {
-    return pr ? this._getAllGitHub(coordinates, pr) : this._getAllLocal(coordinates)
-  }
-
-  async _getAllLocal(coordinates) {
-    await this.ensureCurations()
-    const filePath = `${this.tempLocation.name}/${this.options.repo}/${this._getSearchRoot(coordinates)}.yaml`
-    let result
-    try {
-      result = yaml.safeLoad(fs.readFileSync(filePath))
-    } catch (error) {
-      if (error.code === 'ENOENT') return
-      throw error
-    }
-    const res = await this._getLocalSha(filePath)
-    set(result, '_origin', { sha: res.split(' ')[1], enumerable: false })
-    return result
-  }
-
-  async _getLocalSha(filepath) {
-    const parent = path.dirname(filepath)
-    const command = `cd ${parent} && git ls-files -s ${path.basename(filepath)}`
-    return new Promise((resolve, reject) => {
-      exec(command, (error, stdout) => {
-        if (error) return reject(error)
-        resolve(stdout)
-      })
-    })
+  async getAll(coordinates, pr = null) {
+    // Check to see if there is content for the given coordinates
+    const path = this._getCurationPath(coordinates)
+    const { owner, repo } = this.options
+    const branch = await this.getBranchAndSha(pr)
+    const branchName = branch.sha || branch.ref
+    const url = `https://raw.githubusercontent.com/${owner}/${repo}/${branchName}/${path}`
+    const content = await requestPromise({ url, method: 'GET', resolveWithFullResponse: true, simple: false })
+    if (content.statusCode !== 200) return null
+    // If there is content, go and get it. This is a little wasteful (two calls) but
+    // a) the vast majority of coordinates will not have any curation
+    // b) we need the sha of the curation to form part of the final definition tool chain
+    // At some point in the future we can look at caching and etags etc
+    return this._getFullContent(coordinates, branch.ref)
```

Comment:
should this be `return this._getFullContent(coordinates, branchName)` ?

so we try sha first?

Reply:
We only use sha. By this time we either have the SHA of master or of the PR.